.github/actions/setup-maven-cache/action.yaml

```diff
@@ -83,9 +90,10 @@ runs:
         key: ${{ env.MAVEN_CACHE_KEY }}
         restore-keys: |
           ${{ env.MAVEN_CACHE_KEY_PREFIX }}
+
     - name: Restore maven cache
-      # Restore cache (but don't save it) if we're not on main or stable/* branches
-      if: ${{ !(startsWith(github.ref_name, 'stable/') || github.ref_name == 'main') }}
+      # Restore cache (but don't save it) if we're not on main or stable/* branches and cache is not disabled
+      if: ${{ !(startsWith(github.ref_name, 'stable/') || github.ref_name == 'main') && steps.is-cache-disabled.outputs.is-cache-disabled == 'false' }}
```

Comment:
> and cache is not disabled

❓ Did you consider naming the helper action and variables to avoid the double-negation? So have an `is-cache-enabled` action and check for `steps.is-cache-disabled.outputs.is-cache-disabled == 'true'` to avoid mental complexity of double negation ("So this work when the cache **is not** disabled aka it **is** enabled...").

If possible I'd go for an `is-cache-enabled` action.